src/library/CMakeLists.txt

```diff
@@ -59,89 +56,80 @@ if(F3D_MODULE_ALEMBIC)
-  vtkF3DPostProcessFilter.cxx
-  vtkF3DRenderPass.cxx
-  vtkF3DRenderer.cxx
-  vtkF3DRendererWithColoring.cxx
-  F3DAnimationManager.cxx
-  F3DLoader.cxx
-  F3DLog.cxx
-  F3DOffscreenRender.cxx
-  F3DOptions.cxx
-  readers/F3DReader.cxx
-  readers/F3DReaderFactory.cxx
-  ${F3D_ICON_SOURCE}
+  ${CMAKE_CURRENT_SOURCE_DIR}/vtkF3DConsoleOutputWindow.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/vtkF3DGenericImporter.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/vtkF3DInteractorEventRecorder.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/vtkF3DInteractorStyle.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/vtkF3DMetaReader.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/vtkF3DObjectFactory.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/vtkF3DOpenGLGridMapper.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/vtkF3DPolyDataMapper.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/vtkF3DPostProcessFilter.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/vtkF3DRenderPass.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/vtkF3DRenderer.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/vtkF3DRendererWithColoring.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/F3DAnimationManager.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/F3DLoader.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/F3DLog.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/F3DOffscreenRender.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/F3DOptions.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/readers/F3DReader.cxx
+  ${CMAKE_CURRENT_SOURCE_DIR}/readers/F3DReaderFactory.cxx
+  ${CMAKE_CURRENT_BINARY_DIR}/F3DIcon.cxx
   ${CMAKE_CURRENT_BINARY_DIR}/Config.cxx
   ${MODULES_SPECIFIC_SOURCE_FILES}
+  ${LIB_PLATFORM_SPECIFIC_FILES}
   )
 
-set(F3D_HEADER_FILES
-  vtkF3DConsoleOutputWindow.h
-  vtkF3DGenericImporter.h
-  vtkF3DInteractorEventRecorder.h
-  vtkF3DInteractorStyle.h
-  vtkF3DMetaReader.h
-  vtkF3DObjectFactory.h
-  vtkF3DOpenGLGridMapper.h
-  vtkF3DPolyDataMapper.h
-  vtkF3DPostProcessFilter.h
-  vtkF3DRenderPass.h
-  vtkF3DRenderer.h
-  vtkF3DRendererWithColoring.h
-  F3DAnimationManager.h
-  F3DLoader.h
-  F3DLog.h
-  readers/F3DReader.h
-  readers/F3DReaderFactory.h
-  ${F3D_ICON_HEADER}
-  ${CMAKE_CURRENT_BINARY_DIR}/readers/F3DReaderInstantiator.h
+# List of headers that will be installed
+set(F3D_PUBLIC_HEADERS
+  ${CMAKE_CURRENT_SOURCE_DIR}/F3DLoader.h
   )
 
-# Force static library when creating a macOS bundle
-set(F3D_LIBRARY_TYPE "")
-if(F3D_MACOS_BUNDLE)
-  set(F3D_LIBRARY_TYPE "STATIC")
-endif()
-
-add_library(libf3d ${F3D_LIBRARY_TYPE}
+add_library(libf3d
   ${F3D_SOURCE_FILES}
-  ${F3D_HEADER_FILES}
-  ${LIB_PLATFORM_SPECIFIC_FILES}
   )
 
-target_include_directories(libf3d PUBLIC
-  ${CMAKE_CURRENT_SOURCE_DIR}
-  ${CMAKE_CURRENT_SOURCE_DIR}/readers
-  ${CMAKE_CURRENT_BINARY_DIR}
-  ${CMAKE_CURRENT_BINARY_DIR}/readers
-  ${MODULES_SPECIFIC_INCLUDE_DIRS}
+target_include_directories(libf3d
+  PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+  PRIVATE
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/readers>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/readers>
+  $<BUILD_INTERFACE:${MODULES_SPECIFIC_INCLUDE_DIRS}>
```

Comment:
what is BUILD_INTERFACE ?

Reply:
https://stackoverflow.com/a/58729533/2609654